### DISCUSSION TITLE
版面分析训练，集成到PPStructure中报错

### DISCUSSION THREAD

**Discussion by qybing** [PaddlePaddle/PaddleOCR#13490] [Q&A]
#### 问题描述 / Problem Description
按照官方版面分析的文档，进行版面分析训练，使用的是中文文档训练的，训练成功后，导出预测权重，在PaddleDetection处使用导出的权重正确吐出结果，但是我使用导出的权重集成到PPStructure报错

#### 运行环境 / Runtime Environment
- OS:ubuntu 20.04
- Paddle:2.6.0.post112
- PaddleOCR:2.8.0

#### 复现代码 / Reproduction Code
ocr_engine = PPStructure(table=True,
                         det_model_dir='./pretrained_model/ch_PP-OCRv4_det_server_infer',
                        layout_model_dir='/xxx/xxx/xx/xxxx/PaddleDetection/output_inference/picodet_lcnet_x1_0_layout',
                         ocr=False, show_log=True, gpu_id=0)

#### 完整报错 / Complete Error Message
  File "/xx/xxx/xx/xxx/PaddleOCR/ppstructure/predict_system.py", line 114, in __call__
    layout_res, elapse = self.layout_predictor(img)
  File "/xxx/xxx/xxx/xx/PaddleOCR/ppstructure/layout/predict_layout.py", line 86, in __call__
    self.predictor.run()
ValueError: (InvalidArgument) The input of Op(Conv) should be a 4-D or 5-D Tensor. But received: input's dimension is -1, input's shape is [].
  [Hint: Expected in_dims.size() == 4 || in_dims.size() == 5 == true, but received in_dims.size() == 4 || in_dims.size() == 5:0 != true:1.] (at ../paddle/phi/kernels/cpu/conv_util.h:137)
  [operator < fused_conv2d_add_act > error]

**Comment by SWHL** [comment]
这个从描述来看，看不出来啥问题。你再仔细debug一下吧

**Reply by qybing** [reply]
layout_res, elapse = self.layout_predictor(img)去做版本识别的时候就报错

**Reply by SWHL** [reply]
嗯嗯，描述问题太模糊了。除非给到我们一个可复现的代码，我们这里才好看是否可以帮忙修改。

**Reply by qybing** [reply]
debug具体到版本分析执行self.predictor.run()的时候就报错，我需要怎么debug

**Reply by qybing** [reply]
是需要把权重给到你们，你们尝试一下吗？

**Reply by SWHL** [reply]
这， 我也不清楚了，毕竟报错环境，你那里最清楚了。
大概思路是先找一个能跑不报错的，比较着来看。

**Reply by SWHL** [reply]
不需要给我们权重，你这个问题比较综合，不完全属于PaddleOCR单个项目。
建议用Yolo系列去训练版面分析，好用一些

**Reply by qybing** [reply]
用官方的权重不报错，但是用我训练出来的权重，执行到self.predictor.run()报错，我对比不出来，前面图片的预处理，我没发现有什么异常，但是我导出的训练权重，在PaddleDetection是正确输出的

**Comment by GreatV** [comment]
参考
- https://github.com/PaddlePaddle/PaddleOCR/discussions/13480#discussioncomment-10133608

**Comment by qybing** [comment]
好的，谢谢，在导出权重的时候加入了-o export.benchmark=True ，可以了

**Comment by qybing** [comment]
请问一下如何确定模型是不是GFL架构的
